читает файл, делает датафрейм, где индекс - id асессора

In [1]:
import pandas as pd
import numpy as np

asessors_df = pd.read_csv('D:\\Desktop\\test_ya\\data_task3.csv', sep='\t').set_index('uid')

добавляет столбцы success (1, если jud совпадает с cjud, 0 - если нет) и tasks (везде 1)

In [2]:
asessors_df['sucсess'] = np.where(asessors_df['jud'] == asessors_df['cjud'], 1, 0)
asessors_df['tasks'] = 1

обобщает результаты у каждого ассесора - выдаёт кол-во его заданий ('tasks') и успешно выполненных заданий ('success')


убирает столбцы 'cjud', 'jud', 'docid'

In [3]:
asessors_df = asessors_df.drop(['cjud', 'jud', 'docid'], axis=1).groupby(asessors_df.index).sum()

создаёт столбец 'rate' - доля успешно выполненных заданий среди всех выполненных заданий

In [4]:
asessors_df['rate'] = asessors_df['sucсess'] / asessors_df['tasks']

сортирует по 'rate'

In [5]:
asessors_df = asessors_df.sort_values(by=['rate'])
print(asessors_df.head(15))

     sucсess  tasks      rate
uid                          
56       175    411  0.425791
3        196    426  0.460094
118      186    391  0.475703
390      198    412  0.480583
234       48     99  0.484848
550       93    128  0.726562
9        313    423  0.739953
335      311    419  0.742243
154      306    405  0.755556
71       328    433  0.757506
221      304    400  0.760000
219      300    394  0.761421
92       319    418  0.763158
412      307    402  0.763682
354      327    427  0.765808


Мы видим, что первые пять человек (id 56, 3, 118, 390, 234) явно хуже всего справились - они хорошо выполнили меньше половины заданий, плюс между 234 и 550 очень большой отрыв (больше чем на 20%).
Здесь считалась доля успешно выполненных заданий для каждого асессора. Задание успешно выполненно, если оценки jud и cjud совпали.
В зависимости от ситуации, можно:
1. Считать и наоборот - рейтинг по доле неуспешно выполненных заданий. 
2. По непосредственному количеству неудач/удач (например, если есть правило "У асессора есть право на N ошибок - вне зависимости от общего количества выполненных заданий")
3. Если их ответы на задания 0 и 1 неравноправны (например, не критично, если асессор ответил 0 вместо 1, но очень плохо, если 1 вместо 0) - тогда нужна другая система для оценки успешности выполнения задания (например, она будет небинарной: 0 при jud = 1 cjud = 0, 1 при jud = 0 cjud = 1, 2 при совпадении оценки; или смотреть задания с cjud 1 и 0 отдельно друг от друга)
